pkg/controller/storagecluster/initialization_reconciler.go

```diff
@@ -125,7 +258,8 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 		},
 	}
 	// OBC StorageClass will be added only if it is not a cloud platform
-	if platform, err := r.platform.GetPlatform(r.client); err == nil && !isValidCloudPlatform(platform) {
+	platform, err := r.platform.GetPlatform(r.client)
+	if initData.Spec.ExternalStorage.Enable || err == nil && (!isValidCloudPlatform(platform)) {
```

Comment:
```suggestion
	// OBC storageclass will be returned only in TWO conditions,
	// a. either 'externalStorage' is enabled
	// OR
	// b. current platform is not a cloud-based platform
	if initData.Spec.ExternalStorage.Enable || err == nil && (!isValidCloudPlatform(platform)) {
```
Can you add some comments on similar lines...

Reply:
SUre, will do